packages/flutter/test/services/haptic_feedback_test.dart

```diff
@@ -18,4 +18,32 @@ void main() {
     expect(log, hasLength(1));
     expect(log.single, isMethodCall('HapticFeedback.vibrate', arguments: null));
   });
+
+  test('Haptic feedback variation tests', () async {
+    final List<MethodCall> log = <MethodCall>[];
+
+    SystemChannels.platform.setMockMethodCallHandler((MethodCall methodCall) async {
+      log.add(methodCall);
+    });
+
+    const List<Function> hapticFunctions = const <Function>[
+      HapticFeedback.lightImpact,
+      HapticFeedback.mediumImpact,
+      HapticFeedback.heavyImpact,
+      HapticFeedback.selectionClick,
+    ];
+    final RegExp functionName = new RegExp(r".*\'(\w+)\'");
+
+    for (int i = 0; i < hapticFunctions.length; i++) {
+      await Function.apply(hapticFunctions[i], null);
+      expect(log, hasLength(i + 1));
+      expect(
+        log.last,
+        isMethodCall(
+          'HapticFeedback.vibrate',
+          arguments: 'HapticFeedbackType.${functionName.firstMatch(hapticFunctions[i].toString()).group(1)}',
```

Comment:
totally soliciting less hacky ways of doing this

Reply:
Simpler and less clever might be write a utility function:
```dart
void expectHapticFeedbackCall(Function hapticFunction, String expectedType) {
     final List<MethodCall> log = <MethodCall>[];
 
     SystemChannels.platform.setMockMethodCallHandler((MethodCall methodCall) async {
       log.add(methodCall);
     });
 
     await hapticFunction();
 
     expect(log, hasLength(1));
     expect(log.single, isMethodCall('HapticFeedback.vibrate', arguments: expectedType));
}
```

Then just call that for each of the functions.

Reply:
done